integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -617,7 +617,13 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
     // by default consider all String cols as dims and if any dictionary include isn't present then
     // add it to noDictionaryDims list. consider all dictionary excludes/include cols as dims
     fields.foreach { field =>
-      if (dictIncludeCols.exists(x => x.equalsIgnoreCase(field.column))) {
+      if (dictExcludeCols.exists(x => x.equalsIgnoreCase(field.column))) {
+        val dataType = DataTypeUtil.getDataType(field.dataType.get.toUpperCase())
+        if (dataType != DataType.DATE) {
```

Comment:
how about timestamp?

Reply:
timestamp can be excluded, as now we are supporting timestamp in both include and exclude

Reply:
Ok, how about double,decimal and float?

Reply:
double decimal and float will throw exception as unsupported datatype, may be we can add date also to that list

Reply:
yes, please add

Reply:
we are already having a whitelist for dictionary_exclude that will filter out date. so this check can be removed